mediathread/assetmgr/views.py

```diff
@@ -448,6 +450,245 @@ def final_cut_pro_xml(request, asset_id):
+                ] = [
+                    {
+                        "value": self.root + '/tag/' + tag[k][
+                            'resource_uri'].rstrip('/'), "type": "uri"}
+                    ]
+            else:
+                a_tag_node[
+                    'http://www.openannotation.org/ns/hasBody'] = [
+                    {"value": self.root +
+                     '/tag/' + tag[k]['name'].rstrip('/'),
+                     "type": "uri"}]
+        except Exception:
+            pass
+        a_tag_node[
+            'http://www.w3.org/1999/02/22-rdf-syntax-ns#type'] = [
+            {"value": "http://www.openannotation.org/ns/Annotation",
+             "type": "uri"}]
+        self.tag_num += 1
+        a_tag_urn = 'urn:mediathread:tag' + str(self.tag_num)
+        self.export[a_tag_urn] = a_tag_node
+
+    def parse_response(self, data):
+        video_node = {}
+        # for the video node
+        video_node[
+            'http://purl.org/dc/terms/title'
+            ] = [{"value": data.get('title'), "type": "literal"}]
+        video_node['http://purl.org/dc/terms/description'] = [
+            {"value": data.get('description'), "type": "literal"}]
+        video_node['http://simile.mit.edu/2003/10/ontologies/artstor#url'] = [
+            {"value": data.get('sources')[
+                                            data.get('primary_type')
+                                        ]['url'], "type": "uri"}]
+        video_node['http://purl.org/dc/terms/source'] = [
+            {"value": data.get('primary_type'), "type": "literal"}]
+        video_node['http://purl.org/dc/terms/date'] = [
+            {"value": data.get('modified'), "type": "literal"}]
+        video_node['http://purl.org/dc/terms/contributor'] = [
+            {"value": data.get('author')['username'], "type": "literal"}]
+        self.export[self.root + data.get('local_url').rstrip('/')] = video_node
+        # for annotation node
+        for n in range(0, data.get('annotation_count')):
+            self.parse_annotations(data, n)
+
+    def get(self, request):
+        self.root = 'http://' + request.get_host()
+        assets = Asset.objects.filter(course=request.course)
+        course = Course.objects.get(id=request.course.id)
+        api_response = []
+        n = 0
+        for course_mem in course.user_set.values():
+            n += 1
+            ar = AssetResource(include_annotations=True)
+            ar.Meta.excludes = ['added', 'modified', 'course', 'active']
+            notes = SherdNote.objects.get_related_notes(
+                                                        assets,
+                                                        course_mem['id'],
+                                                        [course_mem['id']],
+                                                        True
+                                                        )
+
+            api_response += (ar.render_list(request, [course_mem['id']],
```

Comment:
The code here renders the items & notes to json. Then, at Ln 687, the json is rendered into the scalar export format. The scalar export could probably work directly against the item/note objects proper. What's the reasoning behind the interim json step?